cl/citations/match_citations.py

```diff
@@ -139,3 +146,110 @@ def match_citation(citation, citing_doc=None):
+            for cm in citation_matches:
+                # The only clue we have to help us with resolution is the guess
+                # of what the supra citation's antecedent is, so we try to
+                # match that string to one of the known case names of the
+                # already matched opinions.
+                if (
+                    strip_punct(citation.antecedent_guess)
+                    in cm.cluster.case_name_full
+                ):
+                    # Just use the first one found, since the candidates should
+                    # all be identical. If nothing is found, then the supra
+                    # reference is effectively dropped.
+                    matched_opinion = cm
+                    break
+
+        # Likewise, if the citation is a short form citation, try to resolve it
+        # to one of the citations that has already been matched
+        elif isinstance(citation, ShortformCitation):
+            # We first try to match by using the reporter and volume number.
+            # However, because matches made using this heuristic may not be
+            # unique, we then refine by using the antecedent guess and only
+            # accept the match if there is a single unique candidate. This
+            # refinement may still fail (because the guess could be
+            # meaningless), in which case the citation is not resolvable and
+            # is dropped.
+            candidates = []
+            for cm in citation_matches:
+                for c in cm.cluster.citations.all():
+                    if (
+                        citation.reporter == c.reporter
+                        and citation.volume == c.volume
+                    ):
+                        candidates.append(cm)
+
+            candidates = list(set(candidates))  # Remove duplicate matches
+            if len(candidates) == 1:
+                matched_opinion = candidates[0]  # Accept the match!
+            else:
+                refined_candidates = []
+                for cm in candidates:
+                    if (
+                        strip_punct(citation.antecedent_guess)
+                        in cm.cluster.case_name_full
+                    ):
+                        refined_candidates.append(cm)
+
+                if len(refined_candidates) == 1:
+                    matched_opinion = refined_candidates[
+                        0
+                    ]  # Accept the match!
+
+        # If the citation is an id citation, just resolve it to the opinion
+        # that was matched immediately prior (so long as the previous match
+        # was successful).
+        elif isinstance(citation, IdCitation):
+            if was_matched:
+                matched_opinion = citation_matches[-1]
+
+        # If the citation is to a non-opinion document, we currently cannot
+        # match these.
+        elif isinstance(citation, NonopinionCitation):
+            pass
```

Comment:
I think I'd move these two to the top. Start with the easy cases.

Reply:
👍

Reply:
Fixed in c839569

Reply:
Should this first one be a `continue` instead of a `pass`?

Reply:
I don't think so, because I still want to trigger the logic at line 258 that toggles the `was_matched` variable appropriately (which is still within the for loop). I remember I used `continue` before but then encountered a bug, and I think switching to `pass` here fixed that.